### PR TITLE
backend/fix/driver input for vehicle downgrade preference {issue-#3865}

### PR DESCRIPTION
**Pull request by shreyaxh**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

{issue-#3865} When vehicle is assigned to driver, drivers should provide input if they want to take downgraded rides(e.g. SEDAN, HATCHBACK rides while having an SUV vehicle). Default downgrade is not required anymore

{fix} Added a `downgradePreference` when the driver activates the RCstatus of vehicle. 

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## How did you test it?

Not tested yet.

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/dashboard/CommonAPIs/src/Dashboard/ProviderPlatform/Driver.hs [hunk 1]
```diff
@@ -517,7 +517,8 @@ data VerificationStatus = PENDING | VALID | INVALID
 
 data RCStatusReq = RCStatusReq
   { rcNo :: Text,
-    isActivate :: Bool
+    isActivate :: Bool,
+    downgradePreference :: Maybe Bool
```
Comment (0utkarsh) [inline Backend/app/dashboard/CommonAPIs/src/Dashboard/ProviderPlatform/Driver.hs:521]: Can we take 3 flags here instead? SUV might want to downgrade to Sedan but not Hatchback

Comment (shreyaxh) [reply Backend/app/dashboard/CommonAPIs/src/Dashboard/ProviderPlatform/Driver.hs]: Sure, I have made the required changes

**Review by hkmangla — CHANGES_REQUESTED** [review]
Driver already have these options. why it needs to be at RC level?

**Comment by shreyaxh** [comment]
> Driver already have these options. why it needs to be at RC level?

Right now we are giving default downgrade values when driver is registered. Which he can update from profile later, with this approach the driver will be able to choose his `downgradePreference` at the time of RC activation itself. Saving him the extra step of updating it through profile and meanwhile getting downgraded vehicle rides which he may not want to take.

**Comment by shreyaxh** [comment]
Bumping this PR @0utkarsh @hkmangla

**Pull request closed without merging**